### DISCUSSION TITLE
Wayland

### DISCUSSION THREAD

**Discussion by frankzen** [bbidulock/icewm#531] [General]
This is part of an answer in the debian mailing list today about Wayland.

Running a window manager in Xwayland is theoretically possible, but practically it'll be worthless.   > You're stuck using one of the window managers or desktop environments > that has been ported to Wayland.  More precise, "that implement Wayland protocol". And considering a current state of testing, one can choose GNOME, KDE, weston or sway. That's it, unless one builds something else from the source.  That means no openbox, twm, icewm, or your favorite panel (because it's a special extension of a Wayland protocol) if you're planning on using Wayland.

It seems Wayland is still some distance off but I ask the question will then eventually cut off Icewm at the knees?

**Comment by gijsbers** [comment]
That is not clear to me. Currently Wayland can easily be ignored, but
if it ever becomes inescapable we can just make a icewm context, I suppose.
The other options don't look attractive to me.

**Reply by covex** [reply]
Well and now Fedora is aiming to "not use X11" in F40.... There is probably some time left for X11, but not that much.

**Reply by josch** [reply]
Wayland is inescapable depending on your hardware. I own the MNT Reform open hardware laptop. There are several options for the main CPU and GPU but none of them are the classical x86 chips with NVidia or AMD graphics. Instead, the graphics "card" are chips like the Vivante GC7000L or ARM Mali G52 or ARM Mali-G610 which are integrated with the CPU in a single package. So the driver for that is either etnaviv, panfrost or panthor, respectively. I am told that the authors of these drivers do not care much about X11 support. For example, trying to run Xorg with etnaviv works with some tweaks to Xorg.conf but results in artifacts that make it de-facto unusable. This is why, on those platforms, you are stuck with wayland.

For those who want to use icewm on these platforms, the choice is to either add wayland support to icewm or to fix X11 support in the etnaviv, panfrost and panthor drivers. The latter didn't happen even though icewem is not the only application which would benefit from that.

In that context, maybe projects like these become useful to enable wayland support for icewem with minimal effort: https://gitlab.freedesktop.org/wayback/wayback/

**Reply by gijsbers** [reply]
> Well and now Fedora is aiming to "not use X11" in F40.... There is probably some time left for X11, but not that much.

He wrote two years ago, but [that still didn't happen](https://www.reddit.com/r/Fedora/comments/1jv1fch/fedora_workstation_42_has_finally_removed_x11/).

> not the only application which would benefit from that.

Indeed, there are 180 X11 window managers.  Like IceWM, they not only are built on X11, but also assume that clients conform to the ICCCM and EWMH standards, which don't exist on Way. [Way breaks a lot](https://gist.github.com/probonopd/9feb7c20257af5dd915e3a9f2d1f2277).

**Reply by JotaRandom** [reply]
and now GNOME, Ubuntu and Fedora dropped X11... looks like "it ever become inescapable".

maybe using [wlroots](https://gitlab.freedesktop.org/wlroots/wlroots) and or the [wlr-protocols](https://gitlab.freedesktop.org/wlroots/wlr-protocols) may help.

**Comment by gijsbers** [comment]
Unfounded gossip. Easy to refute. Just another waste of time, *sigh*...

**Comment by plomari** [comment]
I have given up on Wayland. It's a step backwards in usability, and has advantages only for things that could be done in X11 too. But evil companies like Red Hat plan to drop X11 completely and to go wayland-only (see comments above), so eventually something needs to be done to assure X and icewm can still be used on the Linux desktop in the coming decades. One has to expect that Xorg-on-hardware will get dropped (already the case for some newer hardware, see comments above) or that X11-as-session support will get dropped, so maybe only XWayland will keep working.

XWayland proves that X11 clients (and in fact the Xorg server) can be nested in wayland. Now you'd only need a wayland compositor, which has first class X11 window manager support, and which pretends to the X11 side that wayland clients are X11 clients. Then any X11 window manager could just work? Wouldn't that be, in theory, the way to go?

**Reply by JotaRandom** [reply]
About the nested XWayland, may be a good idea to add the needed information to run iceWM with this nested xwayland in the webpage or the Readme.md (or the Arch Linux wiki).

And since either gnome and KDE will remove X11 support and GTK5 and QT7 rumored to follow this too, it may be a good idea to include the information to run iceWM in this XWayland nested thing since Wayland is definitely here to stay and there is no iceWM for Wayland on the horizon.

**Comment by NexusSfan** [comment]
I mean, there's already XLibre, since Xorg is no longer getting development. Developing a WM for X11 is much easier and smaller than Wayland which is another reason why I use IceWM on XLibre.

**Comment by muezabdalla** [comment]
hello everyone
there is a way to make a wayland window manager without making the compositor by using river. here is a list of window managers made with river:https://codeberg.org/river/wiki/src/branch/main/pages/wm-list.md

you can also read this article for more info:https://isaacfreund.com/blog/river-window-management/

@gijsbers do you think it will be less waste of time now?

**Reply by bojogon7** [reply]
Seem like the best solution
